applications/ChimeraApplication/custom_strategies/strategies/fs_strategy_for_chimera.h

```diff
@@ -0,0 +1,762 @@
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override {}
+
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+
+protected:
+
+    ///@name Protected Life Cycle
+    ///@{
+
+
+    ///@}
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+
+   void SetActiveStateOnConstraint(const Flags& TheFlagToSet ,const bool ValToSet)
+    {
+        ModelPart& rModelPart = BaseType::GetModelPart();
+#pragma omp parallel
+        {
+            ModelPart::MasterSlaveConstraintIteratorType constraints_begin;
+            ModelPart::MasterSlaveConstraintIteratorType constraints_end;
+            OpenMPUtils::PartitionedIterators(rModelPart.MasterSlaveConstraints(),constraints_begin,constraints_end);
+
+            for ( ModelPart::MasterSlaveConstraintIteratorType itConstraint = constraints_begin; itConstraint != constraints_end; ++itConstraint )
+            {
+                if (itConstraint->Is(TheFlagToSet))
+                    itConstraint->Set(ACTIVE, ValToSet);
+            }
+        }
+    }
+
+    double SolveStep() override
+    {
+
+        double start_solve_time = OpenMPUtils::GetCurrentTime();
+        ModelPart& r_model_part = BaseType::GetModelPart();
+
+        // 1. Fractional step momentum iteration
+        r_model_part.GetProcessInfo().SetValue(FRACTIONAL_STEP,1);
+
+        bool converged = false;
+        int Rank = r_model_part.GetCommunicator().MyPID();
+
+        // Activate Constraints for VELOCITY and deactivate PRESSURE
+        SetActiveStateOnConstraint(FS_CHIMERA_VELOCITY_CONSTRAINT, true);
+        SetActiveStateOnConstraint(FS_CHIMERA_PRESSURE_CONSTRAINT, false);
+
+        for(std::size_t it = 0; it < BaseType::mMaxVelocityIter; ++it)
+        {
+            KRATOS_INFO("FRACTIONAL STEP :: ")<<it+1<<std::endl;
+            // build momentum system and solve for fractional step velocity increment
+            r_model_part.GetProcessInfo().SetValue(FRACTIONAL_STEP,1);
+            double norm_dv = BaseType::mpMomentumStrategy->Solve();
+
+            // Check convergence
+            converged = BaseType::CheckFractionalStepConvergence(norm_dv);
+
+            if (converged)
+            {
+                KRATOS_INFO_IF("FSStrategyForChimera ", BaseType::GetEchoLevel() > 0 && Rank == 0)<<
+                    "Fractional velocity converged in " << it+1 << " iterations." << std::endl;
+                break;
+            }
+        }
+
+        // Activate Constraints for PRESSURE and deactivate VELOCITY
+        SetActiveStateOnConstraint(FS_CHIMERA_VELOCITY_CONSTRAINT, false);
+        SetActiveStateOnConstraint(FS_CHIMERA_PRESSURE_CONSTRAINT, true);
+
+        KRATOS_INFO_IF("FSStrategyForChimera ", (BaseType::GetEchoLevel() > 0 && Rank == 0) && !converged)<<
```

Comment:
rank-check is not needed, only prints on one rank

Reply:
ok !